### PR TITLE
Feature/protocols updates

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR updates all protocols to be in line with the latest generator output.

It also updates some docstrings and api docs.

## Fixes

- #983

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: aea/aea.py [hunk 1]
```diff
@@ -117,13 +118,9 @@ def resources(self) -> Resources:
     @resources.setter
     def resources(self, resources: "Resources") -> None:
         """Set resources."""
+        assert type(resources) is Resources, "Must set resources."
         self._resources = resources
 
-    @property
```
Comment (DavidMinarsch) [inline aea/aea.py]: removed as it is not a public class

File: README.md [hunk 2]
```diff
@@ -63,32 +67,34 @@ The following steps are only relevant if you intend to contribute to the reposit
 
 - Install development dependencies (optionally skipping Lockfile creation):
 
-	  pipenv install --dev --skip-lock
+	    pipenv install --dev --skip-lock
```
Comment (marcofavorito) [inline README.md]: not sure the space is needed.

- Before:

 	  pipenv install --dev --skip-lock

- After:

	    pipenv install --dev --skip-lock

File: README.md [hunk 3]
```diff
@@ -63,32 +67,34 @@ The following steps are only relevant if you intend to contribute to the reposit
...
 
 - Install package in development mode (this step replaces `pip install aea[all]` above):
 
-	  pip install -e .[all]
+	    pip install -e .[all]
```
Comment (marcofavorito) [inline README.md]: see above:

- Before

	  pip install -e .[all]

- After

	    pip install -e .[all]

File: README.md [hunk 4]
```diff
@@ -63,32 +67,34 @@ The following steps are only relevant if you intend to contribute to the reposit
...
+
+(`pip install -e ".[all]"` if you use `zsh` rather than `bash`.)
 
 - To run tests (ensure no oef docker containers are running):
 
-	  tox -e py3.7
+	    tox -e py3.7
```
Comment (marcofavorito) [inline README.md]: see above

File: README.md [hunk 5]
```diff
@@ -63,32 +67,34 @@ The following steps are only relevant if you intend to contribute to the reposit
...
 
 - To run linters (code style checks):
 
-	  tox -e flake8
+	    tox -e flake8
```
Comment (marcofavorito) [inline README.md]: see above

File: README.md [hunk 6]
```diff
@@ -63,32 +67,34 @@ The following steps are only relevant if you intend to contribute to the reposit
...
 
 - To run static type checks:
 
-	  tox -e mypy
+	    tox -e mypy
```
Comment (marcofavorito) [inline README.md]: see above

File: README.md [hunk 7]
```diff
@@ -63,32 +67,34 @@ The following steps are only relevant if you intend to contribute to the reposit
...
 
 - To run black code formatter:
 
-	  tox -e black
+	    tox -e black
```
Comment (marcofavorito) [inline README.md]: see above

File: README.md [hunk 8]
```diff
@@ -63,32 +67,34 @@ The following steps are only relevant if you intend to contribute to the reposit
...
 
 - To run bandit security checks:
 
-	  tox -e bandit-main
-	  tox -e bandit-tests
+	    tox -e bandit-main
+	    tox -e bandit-tests
```
Comment (marcofavorito) [inline README.md]: see above

File: README.md [hunk 9]
```diff
@@ -98,7 +104,7 @@ To amend the docs, create a new documentation file in `docs/` and add a referenc
 
 - Fetch submodules:
 
-	  git submodule sync --recursive && git submodule update --init --recursive
+	    git submodule sync --recursive && git submodule update --init --recursive
```
Comment (marcofavorito) [inline README.md]: see above

File: README.md (hunk 3 above)
Comment (DavidMinarsch) [reply README.md]: Yes that's on purpose :)

File: README.md (hunk 2 above)
Comment (DavidMinarsch) [reply README.md]: It shows as correct in Sublime text ...

File: aea/aea.py [hunk 10]
```diff
@@ -117,13 +118,9 @@ def resources(self) -> Resources:
     @resources.setter
     def resources(self, resources: "Resources") -> None:
         """Set resources."""
+        assert type(resources) is Resources, "Must set resources."
```
Comment (marcofavorito) [inline aea/aea.py]: This is OK, but that implies the question: why don't we check the type also in the constructor?

File: aea/agent.py [hunk 11]
```diff
@@ -183,6 +185,21 @@ def start(self) -> None:
...
+        - calls start() on the liveness, and
+        - enters the agent main loop.
+
+        While the liveness of the agent is not stopped it continues to loop over:
+
+            - increment the tick,
+            - call to act(),
+            - sleep for specified timeout,
+            - call to react(),
+            - call to update().
```
Comment (marcofavorito) [inline aea/agent.py]: fix indentation as the list above

File: README.md (hunk 3 above)
Comment (marcofavorito) [reply README.md]: Can you explain the rationale behind that?

In my opinion, it should be without space. Both for an aesthetic reason and a practical reason. For the latter, if I wanted to copy & paste the command on the terminal, I have to remove the spaces manually.

File: aea/protocols/default/__init__.py [hunk 12]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 # ------------------------------------------------------------------------------
 #
-#   Copyright 2018-2020 Fetch.AI Limited
+#   Copyright 2020 fetchai
```
Comment (marcofavorito) [inline aea/protocols/default/__init__.py]: Why you removed `2018`? And why `fetchai` now is lowercase?
The website uses `Fetch.ai`: https://fetch.ai/

Also, please update LICENSE: https://github.com/fetchai/agents-aea/blob/feature/protocols_updates/LICENSE#L189

File: aea/protocols/default/custom_types.py [hunk 13]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 # ------------------------------------------------------------------------------
 #
-#   Copyright 2018-2020 Fetch.AI Limited
+#   Copyright 2020 fetchai
```
Comment (marcofavorito) [inline aea/protocols/default/custom_types.py]: see above

File: README.md (hunk 2 above)
Comment (marcofavorito) [reply README.md]: Yes it happens, the same for the PyCharm plugin

Comment (marcofavorito) [reply README.md]: see below for the discussion

File: aea/protocols/default/__init__.py (hunk 12 above)
Comment (marcofavorito) [reply aea/protocols/default/__init__.py]: Moreover, if we change the copyright here, shouldn't we change it everywhere else?

Comment (DavidMinarsch) [reply aea/protocols/default/__init__.py]: These are the autogenerated protocols. We cannot have a special rule for fetchai, we can in the midterm improve the generator to have a better representation.

File: aea/aea.py (hunk 10 above)
Comment (DavidMinarsch) [reply aea/aea.py]: Fair point!

**Pull request merged**